doc/building-and-running.md

```diff
@@ -80,32 +82,66 @@ kubectl create -f kubernetes/generated/local/
 
-This even works in `magda-combined-db` and `magda-elastic-search`, so you can start up the database (on Minikube), the registry API, and a connector by executing the following three commands in three separate windows:
+A typical use case would be:
+
+1. Start `magda-combined-db` in `Minikube` using `helm:`
+
+From root level of the project directory:
+```bash
+helm install --name magda deploy/helm/magda -f deploy/helm/minikube-dev.yml --set tags.all=false --set tags.combined-db=true
+```
+
+2. Port forward database service port to localhost so that your local running program (outside the `Kubernetes` cluster in `minikube`) can connect to them:
 
 ```bash
-# these three commands don't terminate, so run them in separate terminals
-cd magda-combined-db && npm run dev
+# Port forward database
+# this command doesn't terminate, so run it in a separate terminal
+kubectl port-forward combined-db-0 5432:5432
+```
+
+3. Start the registry API and a connector (via `npm run dev`) by executing the following two commands in two separate terminal windows:
+
+```bash
+# these two commands don't terminate, so run them in separate terminals
 cd magda-registry-api && npm run dev
 cd magda-ckan-connector && npm run dev -- --config ../deploy/connector-config/data-gov-au.json
 ```
 
 See [connectors](connectors.md) for more detailed information about running connectors.
 
+4. (Optional) If later you wanted to start `magda-elastic-search` as well:
+
+Like `magda-combined-db`, `magda-elastic-search` can only be started in `minikiube` via `helm` rather than `npm run dev`.
+
+You need to upgrade previously installed `helm` chart `magda` to include `magda-elastic-search` component:
+
+```bash
+helm upgrade magda deploy/helm/magda -f deploy/helm/minikube-dev.yml --set tags.all=false --set tags.combined-db=true --set tags.elasticsearch=true
+```
+And then, port forward `elasticsearch` so that you can run other components that may need to connect to `elasticsearch` outside the `minikube`:
+
+```bash
+# Port forward elasticsearch
+# this commands doesn't terminate, so run it in a separate terminal
+kubectl port-forward es-data-0 9300:9300
+```
+You can find more information regarding starting `magda` components via `helm` in `minikube` from [the `Running on Minikube` section below](#running-on-minikube)
+
 ## What do I need to run?
 
 Running individual components is easy enough, but how do we get a fully working system?  It is rarely necessary to run _all_ of MAGDA locally, but various components depend on other components as follows:
 
 | Component | Dependencies |
 | --------- | ------------ |
-| `magda-*-connector` | `magda-registry-api` |
+| `magda-*-connector` | `magda-registry-api`, `magda-authorization-api` |
```

Comment:
Actually I just turned off auth for registry-api in minikube in another PR, so we don't need this once the other PR gets merged.